targets/TARGET_NORDIC/TARGET_MCU_NRF51822/device/TOOLCHAIN_GCC_ARM/TARGET_MCU_NORDIC_32K/NRF51822.ld

```diff
@@ -130,7 +130,7 @@ SECTIONS
 		end = __end__;
 		__HeapBase = .;
 		*(.heap*)
-		. = ORIGIN(RAM) + LENGTH(RAM) - Stack_Size;
+		. += (ORIGIN(RAM) + LENGTH(RAM) - Stack_Size) - .;
```

Comment:
Setting the pointer is much more clear. 
Side note: does that arithmetic operation use signed or unsigned integer math as this may lead to incorrect values if unsigned integer is used and the left operand less than the right operand.